src/snowflake/snowpark/mock/_analyzer.py

```diff
@@ -662,9 +662,30 @@ def do_resolve_with_resolved_children(
             )
 
         if isinstance(logical_plan, TableFunctionJoin):
-            self._conn.log_not_supported_error(
-                external_feature_name="table_function.TableFunctionJoin",
-                raise_error=NotImplementedError,
+            # standard optimizer returns here:
+            # return self.plan_builder.join_table_function(
+            #     self.analyze(
+            #         logical_plan.table_function, df_aliased_col_name_to_real_col_name
+            #     ),
+            #     resolved_children[logical_plan.children[0]],
+            #     logical_plan,
+            #     logical_plan.left_cols,
+            #     logical_plan.right_cols,
+            #     self.session.conf.get("use_constant_subquery_alias", False),
+            # )
+            # func = self.analyze(...)
+            # child = resolved_children[logical_plan.children[0]]
+            # source_plan = logical_plan
+            # left_cols = logical_plan.left_cols
+            # right_cols = logical_plan.right_cols
+            # use_constant_subquery_alias = self.session.conf.get(
+            #    "use_constant_subquery_alias", False
+            # )
+
+            # TODO: Fix for local testing.
```

Comment:
having tickets with the TODOs would make it a lot easier to track them for future work

Reply:
We've been fairly loose about this reflecting the low maturity level of the project. I think the time to ratchet this is up is before we merge into main, which should be the next couple of weeks or so.